### PR TITLE
Move pkg/api/v1/ref -> client-go/tools/reference

### PR DESCRIPTION
**Pull request by sttts**
`pkg/api/v1/ref` is the only remaining package copied from pkg/api/v1 to client-go via staging/copy.sh.

### CODE REVIEW

**Comment by sttts** [comment]
/assign @smarterclayton 

Needs approval for pkg/api/v1.

File: staging/src/k8s.io/client-go/tools/reference/ref.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package ref
+package reference
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: @sttts could you move it to client-go/util instead? In my mind, tools/ are for larger features like informers ect. And when k8s.io/pkg is ready, util/ should be moved there.

**Comment by caesarxuchao** [comment]
One nit, otherwise the 4th commit lgtm.

Can we get the 4th commit without the other 3 commits?

File: staging/src/k8s.io/client-go/tools/reference/ref.go (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: util is meant to be independent of kube types, isn't it?

**Comment by sttts** [comment]
> Can we get the 4th commit without the other 3 commits?

done. Wrong rebase target.

File: staging/src/k8s.io/client-go/tools/reference/ref.go (hunk 1 above)
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: Yes. My understanding is that there will be k8s.io/util for non-kube things, and k8s.io/pkg that can import kube types. So how about this reference package to k8s.io/client-go/pkg for now, and move it to k8s.io/pkg in the future?

Comment (sttts) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: I would like to get rid of the pkg dir. Not much sense to have a single helper package there. Why not tools?

Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: In my mind tools will be for features that deserve a README package and an example in /examples. It the reality is move further away from this picture.

Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: To keep us on the same page: we still copy the clientset/listers/informers from kubernetes to client-go. I'm working on PRs to remove the copies in kubernetes.

Comment (sttts) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: Do you have anything in the works already? Wanted to tackle this as well very soon.

Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/tools/reference/ref.go]: Yes, will send a PR today.

**Pull request merged**